### ISSUE TITLE
Uncaught ReferenceError with deno bundle >= 1.5.0

### ISSUE THREAD

**Issue by c4spar** [denoland/deno#9467]
I have another bundle issue where an Uncaught ReferenceError occurred.

The error occurs with all deno versions >= 1.5.0. (I am on 1.7.2). With deno <= 1.4.6 it works.

```typescript
import { tty } from "https://deno.land/x/cliffy@v0.17.2/ansi/tty.ts";
import { eraseDown } from "https://deno.land/x/cliffy@v0.17.2/ansi/ansi_escapes.ts";

eraseDown();
tty.eraseDown();
```

```
error: Uncaught ReferenceError: Cannot access 'mod' before initialization
    const methodList = Object.entries(mod);
```

The bundled code has the wrong order. I think this is because the tty module also uses the ansi_escape module internally. If i import the ansi_escape module before the tty module, everything works:

```typescript
import { eraseDown } from "https://deno.land/x/cliffy@v0.17.2/ansi/ansi_escapes.ts";
import { tty } from "https://deno.land/x/cliffy@v0.17.2/ansi/tty.ts";

eraseDown();
tty.eraseDown();
```

I think this issue could be related to https://github.com/denoland/deno/issues/8211.

**Comment by lucacasonato** [comment]
@c4spar Can you try again with the latest canary? (`deno upgrade --canary`)

**Comment by c4spar** [comment]
@lucacasonato it's also not working with the latest canary `1.7.2+723c99d`.

**Comment by c4spar** [comment]
Here is a smaller reproducable example:
```typescript
// main.ts
import {bar} from "./bar.ts";
import {foo} from "./foo.ts";

console.log(foo());
console.log(bar);
```
```typescript
// foo.ts
export function foo(): string {
    return `foo`;
}
```
```typescript
// bar.ts
import * as foo from "./foo.ts";

export const bar = factory();

function factory() {
    return foo;
}
```
```
$ deno bundle --unstable main.ts bundle.js
$ deno run --unstable ./bundle.js
error: Uncaught ReferenceError: Cannot access 'mod' before initialization
    return mod;
    ^
    at factory (file:///bundle.js:11:5)
    at file:///bundle.js:1:13
```
```typescript
// bundle.ts
const bar = factory();
const mod = function() {
    function foo() {
        return `foo`;
    }
    return {
        foo
    };
}();
function factory() {
    return mod;
}
console.log(foo());
console.log(bar);
```

**Comment by c4spar** [comment]
The problem is the `import * as`.

If i change the content of `bar.ts` to:
```typescript
// bar.ts
import {foo} from "./foo.ts";

export const bar = factory();

function factory() {
    return {foo};
}
```
it works.

```
// bundle.ts
function foo() {
    return `foo`;
}
const bar = factory();
function factory() {
    return {
        foo: foo
    };
}
console.log(foo());
console.log(bar);
```

**Comment by bartlomieju** [comment]
Fixed in 1.8
